content/en/blog/_posts/2025-01-14-devices-failure-handling/index.md

```diff
@@ -0,0 +1,403 @@
+from [Sergey Kanzhelev and Mrunal Patel's talk at KubeCon NA
+2024](https://sched.co/1i7pT). You can follow the links to
+[slides](https://static.sched.com/hosted_files/kccncna2024/b9/KubeCon%20NA%202024_%20Navigating%20Failures%20in%20Pods%20With%20Devices_%20Challenges%20and%20Solutions.pptx.pdf?_gl=1*191m4j5*_gcl_au*MTU1MDM0MTM1My4xNzMwOTE4ODY5LjIxNDI4Nzk1NDIuMTczMTY0ODgyMC4xNzMxNjQ4ODIy*FPAU*MTU1MDM0MTM1My4xNzMwOTE4ODY5)
+and
+[recording](https://www.youtube.com/watch?v=-YCnOYTtVO8&list=PLj6h78yzYM2Pw4mRw4S-1p_xLARMqPkA7&index=150).
+
+## The AI/ML boom and its impact on kubernetes
+
+The rise of AI/ML workloads has brought new challenges to Kubernetes. These
+workloads often rely heavily on specialized hardware, and any device failure can
+significantly impact performance and lead to frustrating interruptions. As
+highlighted in the recent [Llama
+paper](https://ai.meta.com/research/publications/the-llama-3-herd-of-models/),
+hardware issues, particularly GPU failures, are a major cause of disruption in
+AI/ML training. You can also learn how much effort NVIDIA spend on handing
+devices failures and maintenance in the KubeCon talk by [Ryan Hallisey and Piotr
+Prokop All-Your-GPUs-Are-Belong-to-Us: An Inside Look at NVIDIA's Self-Healing
+GeForce NOW
+Infrastructure](https://kccncna2024.sched.com/event/1i7kJ/all-your-gpus-are-belong-to-us-an-inside-look-at-nvidias-self-healing-geforce-now-infrastructure-ryan-hallisey-piotr-prokop-pl-nvidia)
+([recording](https://www.youtube.com/watch?v=iLnHtKwmu2I)) as they see 19
+remediation requests per 1000 nodes a day!
+
+We also see data centers offering spot consumption models and overcommit on
+power, making device failures commonplace and a part of the business model.
+However Kubernetes’s view on resources is still very static. The resource is
+either there or not. And if it is there, the assumption is that it will stay
+there. This assumption combined with the overall complexity of a set up, leads
+to a variety of failure modes, which we discuss here.
+
+### Understanding AI/ML workloads
+
+Generally, all AI/ML workloads need specialized hardware, have challenging
+scheduling requirements, and are expensive when idle. AI/ML workloads typically
+fall into two categories - training and inference. Here is an oversimplified
+view on those categories’ characteristics, which are different traditional,
+workloads like web services:
+
+* **Training:** These workloads are resource-intensive, often consuming entire
+  machines and running as gangs of pods. Training jobs are usually "run to
+  completion" - but that could be days, weeks or even months. Any failure in a
+  single pod can necessitate restarting the entire step across all the pods.  
+* **Inference:** These workloads are usually long-running or run indefinitely,
+  and can be small enough to consume a subset of a Node’s devices or large to
+  span multiple nodes. They often require downloading huge files with the model
+  weights.
```

Comment:
I'd use a description list. If a description list isn't right here, we can even fix the CSS in an enabling PR if that's what it takes.

Reply:
applied. Let's see if it is enabled on this repo